### PR TITLE
Fix compilation warnings

### PR DESCRIPTION
**Pull request by vigoux**
This PR fixes some compilation warnings triggered while compiling the C library.

Some of them are triggered when building within neovim (see https://github.com/neovim/neovim/pull/12321 for more informations).

cc @tjdevries

### CODE REVIEW

**Comment by maxbrunsfeld** [comment]
Thanks @vigoux! This all looks good; sorry that I keep forgetting to add `(void)` in the parameter list when I add new code 😬 . I'll merge when CI goes green.

**Comment by tjdevries** [comment]
Oh, I forgot to ask in our thread, @vigoux do you know which compiler switch throws those warnings? Could potentially add that here if tree-sitter doesn't mind :wink: 

We understand if you don't want to maintain/add these things. Just makes it a lot easier for us to vendor the code into our runtime without having to do a lot of manual patching when updating commits..

**Comment by maxbrunsfeld** [comment]
Yeah, I am fine with adding to my own build whichever GCC flag causes it to catch these issues.

**Comment by vigoux** [comment]
@tjdevries : I thinks that it is just a `-Wstrict-prototypes` flag, but I don't know the compiler though.

**Comment by maxbrunsfeld** [comment]
That doesn't seem to catch any errors on master with Clang, but maybe it would with gcc.

**Comment by vigoux** [comment]
> That doesn't seem to catch any errors on master with Clang, but maybe it would with gcc.

I really don't know which compiler neovim is using, but it should trigger some warnings.

**Comment by maxbrunsfeld** [comment]
If you're interested, as part of this PR you could add a `CFLAGS=<something>` to the `.travis.yml`, in a top level `env` block. But I'm also ok with punting on that for now.

**Comment by vigoux** [comment]
Okay I added the flags to consider warnings errors but now I am regreting, do you want me to fix every single warning along with this PR ?

**Comment by vigoux** [comment]
That was faster than expected, all warnings should be fixed now.
Nothing too difficult.

**Comment by maxbrunsfeld** [comment]
Were there other warnings?

File: .travis.yml [hunk 1]
```diff
@@ -2,6 +2,9 @@ language: rust
 rust:
   - stable
 
+env:
+  CFLAGS="-Wall -Wextra -Werror -Wstrict-prototypes"
```
[the remark this answers is not in the document]
Comment (vigoux) [reply .travis.yml:6]: For `gcc`, `strict-prototypes` is not triggered by `extra` nor `all`.

[the remark this answers is not in the document]
Comment (maxbrunsfeld) [reply .travis.yml:6]: Ah, nevermind. I see from the docs that `Wall` does not really mean "all", and neither does extra. Disregard this.

[the remark this answers is not in the document]
Comment (vigoux) [reply .travis.yml:6]: Yeah those warnings set are strange sometimes, but I think it is quite recommended to use the `-Wall -Wextra -Werror`

**Comment by maxbrunsfeld** [comment]
Thanks for doing this!

**Pull request merged**